src/ngx_http_lua_socket_tcp.c

```diff
@@ -2673,6 +2694,111 @@ ngx_http_lua_socket_tcp_settimeout(lua_State *L)
 }
 
 
+static int
+ngx_http_lua_socket_tcp_set_receive_timeout(lua_State *L)
+{
+    int                     n;
+    ngx_int_t               timeout;
+
+    ngx_http_lua_socket_tcp_upstream_t  *u;
+
+    n = lua_gettop(L);
+
+    if (n != 2) {
+        return luaL_error(L, "ngx.socket set_receive_timeout: expecting at least 2 "
+                          "arguments (including the object) but seen %d",
+                          lua_gettop(L));
+    }
+
+    timeout = (ngx_int_t) lua_tonumber(L, 2);
+
+    lua_rawseti(L, 1, SOCKET_TIMEOUT_INDEX);
```

Comment:
@agentzh, I'm looking at the use of `SOCKET_TIMEOUT_INDEX` in order to revise that. It seems that the line `lua_rawgeti(L, 1, SOCKET_TIMEOUT_INDEX)` appears only when using `connect()`, but it is used anyway. I think it should be used on `connect` only when creating a new `ngx_http_lua_socket_tcp_upstream_t` object, meaning in line 573. In that case, I think it would be correct to split `SOCKET_TIMEOUT_INDEX` so that there would be an index for each of the connect, read and write actions. Do you agree?